### DISCUSSION TITLE
Sorting articles by weight and then date

### DISCUSSION THREAD

**Discussion by jromal** [Lednerb/bilberry-hugo-theme#474] [Q&A] [answered]
Hi all.

I have a page with only articles. No pages, nothing else than articles. 
I may add some pages later. But this is not relevant.

On the overview of articles I want to sort the articles by a different criteria than date. 
I want some top articles be sorted by a weight value (and then date). But it is sorting only by date.

I have seen a discussion about sorting pages by weight on the top navigation bar.
But I am looking for a sorting on the overview. I have tried to modify some partials but without success.
I have used this, but I am not sure if it is right and where to put it.
`{{range .Pages.GroupBy "Weight" "desc"}}`

Does someone know how could this be done and can point me out how to do it?
What is the partial file to modify that content (the overview of articles, that is the only type of object I use)?

Many thanks.

**Comment by igor-baiborodine** [comment, marked as answer]
@jromal, your `GroupBy` should go to this line: 
https://github.com/igor-baiborodine/bilberry-hugo-theme-sandbox/blob/d00660355b7246281d9212c8afc34a8a3089cd81/layouts/index.html#L27

And this article may help you with proper grouping: 
https://doray.me/today-i-learned/hugo/how-to-render-grouped-articles/

**Comment by jromal** [comment]
Many thanks.

At the end nothing extra was needed. I interpreted the value of weight in the wrong way.
The lower the value the higher the relevance on the list. 

I found it on the official documentation. https://gohugo.io/templates/lists/#order-content
